ape_etherscan/exceptions.py

```diff
@@ -31,14 +31,14 @@ def __init__(self, response: Response):
 
 def get_request_error(response: Response) -> EtherscanResponseError:
     response_data = response.json()
-    if "result" in response_data:
+    if "result" in response_data and response_data["result"]:
         message = response_data["result"]
     elif "message" in response_data:
         message = response_data["message"]
     else:
         message = response.text
 
-    if "max rate limit reached" in message.lower():
+    if "max rate limit reached" in response.text.lower():
```

Comment:
Random thought: can ape handle the rate limit exceeded error and progressively slow down requests?

Reply:
Yes and no. Currently, there is an exception that gets raised that is easy to catch and the user can wait on their own.

We could add a config for this though, perhaps? `request_retry_wait_time` or something